pypeit/spectrographs/spectrograph.py

```diff
@@ -278,20 +278,25 @@ def get_datasec_img(self, filename, det=1, force=True):
             # Get the image shape
             raw_naxis = self.get_raw_image_shape(filename, det=det)
 
-            binning = self.get_meta_value(filename, 'binning')
-#            binning = self.parse_binning(filename)
+            binning_pypeit = self.get_meta_value(filename, 'binning')
 
             data_sections, one_indexed, include_end, transpose \
                     = self.get_image_section(filename, det, section='datasec')
+            if transpose:
```

Comment:
Please add a comment on what `binning_raw` is.
Same with `binning_pypeit`.

Reply:
note that it had been my intent to have binning *always* be (spec,spat)
anywhere in the code.  That is, if it is written as (spat,spec) to the header
or whatever, it is *immediately* flipped so that any/every variable will have it 
in that manner.  Can we maintain this?

Reply:
Docs added. Take a look at what I wrote there. @kbwestfall and I thought through this quite carefully. The preferred solution is to totally nuke processimages and start from scratch since nearly all of the code it wraps is incoherent garbage. The current stopgap is this fix, which was done thoughtfully to not interfere with sec2slice does, which works in the fits convention.

Reply:
I don't really agree with the sentiment of Joe's third sentence, but let me explain the thinking on binning.  The issue is primarily that we want to be able to have the data/overscan sections defined either directly in `spectrograph.detector` or read from the fits headers.  To keep things consistent, it's important then to have, e.g., `spectrograph.detector[0]['datasec']` be ordered as it would be read via a FITS header.  Depending on the setup of the given spectrograph, `spectrograph.get_image_section` then provides the desired section and, based on the spectral axis defined the instrument, whether or not that section should be transposed to match the way that `spectrograph.load_raw_frame` orients the data array.  An alternative approach to what we've already done is to force `get_image_section` to always return (spec,spat).  This isn't something Joe and I talked about yesterday, but it may mean we don't need to use/return `transpose`.  The only thing I insisted on was that `parse.sec2slice` should never transpose the section with respect to the provided binning or vice versa.  The order of the section should be identical to the order of binning when passed to that function.

Reply:
Am a bit confused, but I'm convinced you two came to a good solution.